examples/threading/threading.c

```diff
@@ -2,32 +2,58 @@
 #include <unistd.h>
 #include <stdlib.h>
 #include <stdio.h>
+#include <time.h>
+#include <errno.h>
+#include <string.h>
 
 // Optional: use these functions to add debug or error prints to your application
 #define DEBUG_LOG(msg,...)
 //#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
 #define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
 
-void* threadfunc(void* thread_param)
+void* threadfunc(void* tdat_v)
 {
+	struct thread_data* tdat = (struct thread_data*) tdat_v;
+	tdat->thread_complete_success = true;
```

Comment:
@georgehodgkins consider whether you should set thread_complete_success here, since your thread hasn't completed with success.  It would be a problem if someone else uses `tdat_v` in a future design/implementation.